packages/dev/core/src/Meshes/Node/Blocks/linearClonerBlock.ts

```diff
@@ -0,0 +1,213 @@
+import { NodeGeometryBlock } from "../nodeGeometryBlock";
+import type { NodeGeometryConnectionPoint } from "../nodeGeometryBlockConnectionPoint";
+import { RegisterClass } from "../../../Misc/typeStore";
+import { NodeGeometryBlockConnectionPointTypes } from "../Enums/nodeGeometryConnectionPointTypes";
+import { Matrix, Quaternion, Vector3 } from "../../../Maths/math.vector";
+import { type NodeGeometryBuildState } from "../nodeGeometryBuildState";
+import type { VertexData } from "core/Meshes/mesh.vertexData";
+import { type Nullable } from "core/types";
+import { PropertyTypeForEdition, editableInPropertyPage } from "core/Decorators/nodeDecorator";
+
+/**
+ * Block used to clone geometry along a line
+ */
+export class LinearClonerBlock extends NodeGeometryBlock {
+    private _vertexData: VertexData;
+    private _currentIndex: number;
+    private _currentLoopIndex: number;
+
+    /**
+     * Gets or sets a boolean indicating that this block can evaluate context
+     * Build performance is improved when this value is set to false as the system will cache values instead of reevaluating everything per context change
+     */
+    @editableInPropertyPage("Evaluate context", PropertyTypeForEdition.Boolean, "ADVANCED", { notifiers: { rebuild: true } })
+    public evaluateContext = true;
+
+    /**
+     * Gets the current index in the current flow
+     * @returns the current index
+     */
+    public getExecutionIndex(): number {
+        return this._currentIndex;
+    }
+
+    /**
+     * Gets the current face index in the current flow
+     * @returns the current face index
+     */
+    public getExecutionFaceIndex(): number {
+        return 0;
+    }
+
+    /**
+     * Gets the current loop index in the current flow
+     * @returns the current loop index
+     */
+    public getExecutionLoopIndex(): number {
+        return this._currentLoopIndex;
+    }
+
+    /**
+     * Create a new LinearClonerBlock
+     * @param name defines the block name
+     */
+    public constructor(name: string) {
+        super(name);
+        this.registerInput("geometry", NodeGeometryBlockConnectionPointTypes.Geometry, false);
+        this.registerInput("count", NodeGeometryBlockConnectionPointTypes.Int, true, 1, 0);
+        this.registerInput("origin", NodeGeometryBlockConnectionPointTypes.Vector3, true, new Vector3(0, 0, 0));
+
+        //Direction is magnitude per step, or total distance
+        this.registerInput("direction", NodeGeometryBlockConnectionPointTypes.Vector3, true, new Vector3(1, 0, 0));
+
+        //rotation is magnitude per step, or total rotation
+        this.registerInput("rotation", NodeGeometryBlockConnectionPointTypes.Vector3, true, new Vector3(0, 0, 0));
+
+        //scale is magnitude per step, or total rotation
+        this.registerInput("scale", NodeGeometryBlockConnectionPointTypes.Vector3, true, new Vector3(0, 0, 0));
+
+        this.registerOutput("output", NodeGeometryBlockConnectionPointTypes.Geometry);

```

Comment:
to be flexible we accept float for scaling:
https://github.com/BabylonJS/Babylon.js/blob/58a04329c4d1f591d4c783c694541a7dff0e62e6/packages/dev/core/src/Meshes/Node/Blocks/Instances/instantiateBlock.ts#L40